### PR TITLE
Added content to 'Forking' and 'Pull Requests' sections

### PR DESCRIPTION
**Pull request by phschimm**
Minor adjustments:
* added inter-document cross references
* fixed some spelling and grammar errors
* incorporated fixes made by @lmarniazman in [PR #6](https://github.com/devonfw/.github/pull/6)

### CODE REVIEW

File: CONTRIBUTING.asciidoc [hunk 1]
```diff
@@ -49,89 +49,98 @@ The major technology stacks have their own repository carrying the prefix `devon
 Tools we provide typically have a repository named like the tool they provide (omitting prefixes like `devon[fw]`):
 
 * https://github.com/devonfw/tools-cobigen[CobiGen] (incremental code generator)
-* https://github.com/devonfw/ide[devon-ide] (development environment)
+* https://github.com/devonfw/ide[devonfw IDE] (development environment)
```
Comment (hohwille) [inline CONTRIBUTING.asciidoc]: Thanks for all your improvements, can you write the official name: `devonfw-ide`

File: CONTRIBUTING.asciidoc [hunk 2]
```diff
@@ -49,89 +49,98 @@ The major technology stacks have their own repository carrying the prefix `devon
...
 * https://github.com/devonfw/sonar-devon4j-plugin[sonar-devon4j-plugin] (SonarQube plugin for architecture validation)
 * https://github.com/devonfw/cicdgen[CICDgen] (generator for CI/CD)
-* https://github.com/devonfw/devon-docgen[docgen] (generator for AsciiDoc to PDF and other formats)
-* https://github.com/devonfw-forge/solicitor[solicitor] (License compatibility checker)
+* https://github.com/devonfw/devon-docgen[DocGen] (generator for AsciiDoc to PDF and other formats)
```
Comment (hohwille) [inline CONTRIBUTING.asciidoc]: https://github.com/devonfw/docgen[DocGen]

File: CONTRIBUTING.asciidoc [hunk 3]
```diff
@@ -49,89 +49,98 @@ The major technology stacks have their own repository carrying the prefix `devon
...
+* Check out your fork's master branch: `git checkout master` (assuming you're working on a feature branch)
+* Merge the changes from upstream/master into your fork's master branch: `git merge upstream/master`
+* This brings your fork's master branch into sync with the original repository without losing changes on your local feature branch.
+* Switch back to your feature branch to continue work: `git checkout «feature-branch»`
 
 === Code Changes
 Before you start with your code changes, please check the following conventions:
 
-* For each programming language we have a stack repository (see above) containing documentation about the coding conventions (example: https://github.com/devonfw/devon4j/blob/develop/documentation/coding-conventions.asciidoc[Java]). Please read and follow this conventions before doing (bigger) changes.
+* For each programming language we have a stack repository (see <<Repositories>>) containing documentation about the coding conventions (example: https://github.com/devonfw/devon4j/blob/develop/documentation/coding-conventions.asciidoc[Java]). Please read and follow these conventions before making (bigger) changes.
```
Comment (hohwille) [inline CONTRIBUTING.asciidoc]: see xref:repositories[Repositories]

**Comment by phschimm** [comment]
> We also discussed to move the "Getting started with GitHub" documentation out of this guide to make it more compact so it would be a separate asciidoc file linked from here. Can you file an issue for this as well so we do not forget about this...

Sure, which sections of the document would this pertain?  
`Account`, `Organizations` and `Repositories` maybe?

File: CONTRIBUTING.asciidoc (hunk 2 above)
Comment (phschimm) [reply CONTRIBUTING.asciidoc]: GitHub seems to mangle this link, but in AsciiDoc it should work correctly.
Should this be called "devonfw-docgen" or something else, or ist "DocGen" fine?

File: CONTRIBUTING.asciidoc [hunk 4]
```diff
@@ -147,38 +156,41 @@ After that a `git push` is sufficient.
...
+=== Pull Requests
+Once you have completed your changes and DoD, you can finally https://help.github.com/en/articles/creating-a-pull-request[create] a https://help.github.com/en/articles/about-pull-requests[pull request] (PR).
 
 Please ensure the following aspects:
 
-* Follow the same conventions as for commit messages (see above) also for the title of the pull-request
-* In case you are providing a https://help.github.com/en/articles/about-pull-requests[PR] that is not yet ready for merging, please prefix the title with `WIP:`. This can be helpful to get feedback from https://travis-ci.com[travis] or start early discussions on complex changes. Once you reached a state ready to be merged, simple edit the title and remove the `WIP:` prefix.
-* Also add the issue(s) to the description of the pull-request (e.g. `fixes #«issue-id»`).
-* Finally, click on the green `Create Pull Request` button.
+* When selecting a title for your pull request, follow the same conventions that apply to xref:Committing Changes[commit messages].
```
Comment (hohwille) [inline CONTRIBUTING.asciidoc]: `xref:committing-changes[commit messages]`

**Pull request merged**